lib/shopify_api/only_send_mutated_params.rb

```diff
@@ -17,6 +10,13 @@ def method_missing(method_symbol, *args)
       super
     end
 
+    def save
+      slice_attributes_on_changed
+      base = super
+      clear_changed_attributes
```

Comment:
actually a good case to handle here would be that if the transaction is unsuccessful that we restore the original attributes to their before sliced state